### ISSUE TITLE
Stop rendering generic landuse=military

### ISSUE THREAD

**Issue by imagico** [gravitystorm/openstreetmap-carto#2670]
I recently looked a bit at use of `landuse=military` worldwide and i suggest we stop rendering generic areas with this tag, i.e. ones that have no supplemental `military=*` tag.

The reason is the tag is used for a large number of very different things many of which have very little to do with the main purpose of the tag (that is to indicate areas used primarily for military purposes - see [wiki](https://wiki.openstreetmap.org/wiki/Tag:landuse%3Dmilitary)).  In particular the tag is widely used to mark exclusion zones of any kind while many real military landuses are not access restricted at all.

There are also (esp. in the US) military areas inside military areas (like an airfield or barracks inside a military base) which makes no sense at all the way we render it.

We should continue to render `military=danger_area` - though probably in a different styling and we could render additional `military=*` values which are common and consistently used.  We could also render access restrictions on `landuse=military` or `boundary=exclusion_zone` although neither of these is currently used in significant  numbers.

**Comment by dieterdreist** [comment]
sent from a phone

> On 2. Jul 2017, at 22:59, Christoph Hormann <notifications@github.com> wrote:
> 
> while many real military landuses are not access restricted at all.


really? Can you give some examples?

**Comment by imagico** [comment]
In Germany many military training areas (Truppenübungsplatz) are at least partly open to the general public - you can see this often in the map due to the lack of access restrictions on roads.  AFAIK in the US there are also many military bases where public roads cross with no restrictions.

I don't want to get into tagging discussion here but it is essential to separate access restrictions from military use of an area - these are two different concepts which might frequently co-exist but one does not necessarily imply the other (in either direction).  The way we render this implying access restrictions does not help in that regard and use of tags shows that.

**Comment by dieterdreist** [comment]
sent from a phone

> On 3. Jul 2017, at 10:22, Christoph Hormann <notifications@github.com> wrote:
> 
> In Germany many military training areas (Truppenübungsplatz) are at least partly open to the general public - you can see this often in the map due to the lack of access restrictions on roads. AFAIK in the US there are also many military bases where public roads cross with no restrictions.


I see what you are after: accessible roads crossing otherwise inaccessible areas, right? Because I don't think you could stop your car and go for a walk if you're on one of those roads that cross a Truppenübungsplatz. military landuse almost always implies access=no.

**Comment by imagico** [comment]
No, but as i said i don't want to turn this into a tagging discussion.

The point of this issue is that landuse=military is used for very different things with very different practical meaning for the map user and we should not render all of this in a unified way implying that it is all the same.

**Comment by dieterdreist** [comment]
sent from a phone

> On 3. Jul 2017, at 17:03, Christoph Hormann <notifications@github.com> wrote:
> 
> The point of this issue is that landuse=military is used for very different things with very different practical meaning for the map user and we should not render all of this in a unified way implying that it is all the same.

that's what you said, my point was that military landuse generally implies non-accessibility to the public or at least access regulations, while there might be a few exceptions like public roads crossing training areas. Still I believe we should keep rendering this landuse. Other landuses, e.g. industrial, also can be set on very different areas. Also landuse=forest can mean very different kinds of things and have very different implications for the map user.

**Comment by wmyrda** [comment]
On the map for example we do render buildings if they have just building=yes regardless of their purpose. Additional tagging always helps and should be encouraged in better rendering like building=church. That helps having more complete map. Stopping rendering item at all with proper tag added just not a full set of tags is too drastic move.

military=danger_area is from what I know proper tag for areas that are closed to public only for specific time hence why I added it to the area in Poland http://www.openstreetmap.org/relation/4022746
If you want more military areas tagged that way please change that gender color on it and You will see the difference shortly after.

**Comment by pnorman** [comment]
I'm a bit skeptical, but haven't had a real systematic look at other tags on `landuse=military` stuff.

**Comment by andrzej-r** [comment]
As it is, military areas are rendered way too prominently. Look for example at Cambridge, UK. The riffle range to the west of the city is the single most prominent feature in the whole area, dwarfing the city itself.

My proposal is not to stop rendering it altogether (it is an interesting feature for those nearby) but dial down the rendering. Thin boundary lines and no hatching are IMHO more than enough.

**Comment by SomeoneElseOSM** [comment]
(I'm sure I've said this before somewhere, so apologies for duplication).  
What I did was to make it "a bit less pink" and use only half as many lines:
http://map.atownsend.org.uk/maps/map/map.html#zoom=12&lat=52.2001&lon=0.099

The colour might need tweaking to match OSM Carto's generally stronger colours, but other than that it should work here.
(see https://github.com/SomeoneElseOSM/openstreetmap-carto-AJT/commit/48e56466fd6db9570b940a048fb9f35699427fe2 for the actual symbol change)

**Comment by kocio-pl** [comment]
Could somebody make test renderings with the AJT style? I don't have too much time for this and it would be quite easy to start playing with the osm-carto code.

**Comment by kocio-pl** [comment]
The changes to `landuse=military` and `military=danger` rendering have been just merged into master (see  #3057). I think when it's not that prominent, it's still worth showing, no matter what specific function the area plays. Does anybody want to change the rendering of some special subtypes, as suggested in proposition?

**Comment by matthijsmelissen** [comment]
Agree this solves the issue.

**Comment by imagico** [comment]
The situation of how landuse=military is used has not changed significantly from when this issue has been created - use of landuse=military (both in cases where it applies and where it does not) is growing fast so there is a lot of possibility to support correct and meaningful mapping by encouraging mappers to provide more specific and more meaningful information.

How a change in how generic landuse=military is rendered can be assumed to resolve this issue is beyond my comprehension.

**Comment by kocio-pl** [comment]
Because it was possible (for me) that opening this issue has been pushed simply by too prominent rendering.

Also description of the problem is so complex (touching multiple problems), that it's hard for me to guess what's the action plan and when we can say "it's solved" - maybe rendering change is enough, maybe when generic landuse=military is not rendered any more, or maybe it's just a tagging problem?

**Comment by imagico** [comment]
The issue makes a clear suggestion and explains the reasoning behind this.  There is nothing complex about it and no guesswork needed to understand it.

Obviously there is disagreement regarding if what this issue suggests should be implemented but that does not mean doing something else solves this issue.

**Comment by kocio-pl** [comment]
Sorry, it was beyond my comprehension. But now that I'm sure what do you mean, I'm against stopping rendering generic `landuse=military`, hence closing this ticket as declined.

**Comment by polarbearing** [comment]
I'm against stopping rendering generic landuse=military as well. 
I had a quick look at https://taginfo.openstreetmap.org/keys/military#values which is 60% bunker, 10% barracks, and the rest scattered different things. It it not clear to me which of them should support rendering the landuse and which not. 
I agree to close the issue unless we have an analysis which unspecific landuse tagging causes problems. 
landuse=industry can have lots of different forms as well, and we do not require to specify if that is petrol-chemical or computer assembly.

**Comment by Tomasz-W** [comment]
As I see, most of us here is against a change mentioned in title, so I think this issue could be closed.

**Comment by kocio-pl** [comment]
> I agree to close the issue unless we have an analysis which unspecific landuse tagging causes problems.

That's what I would need also to reopen the issue, but that's of course always possible. At this moment I don't understand the specific problems which caused opening this ticket.

**Comment by imagico** [comment]
So @matthijsmelissen thinks this issue is solved, @kocio-pl does not understand it and @polarbearing thinks it should not be solved.

Just trying to understand the reasoning here.

**Comment by matthijsmelissen** [comment]
Sorry for confusion - what I meant is that after #3057, rendering generic landuse=military does not bother me anymore, so for me #3057 solves the current issue. In that sense I agree with @polarbearing.

By the way, I think most examples you give are tagging errors - like using landuse=military inside another landuse=military, or using landuse=military for generic exclusion zones.

**Comment by imagico** [comment]
@matthijsmelissen - do you subscribe to the documented mission of this style to support correct and accurate mapping?  If yes, do you think the way `landuse=military` is currently rendered encourages correct use (i.e. for areas that are actually used for military puposes) and discourages incorrect use (to map access restrictions or exclusion zones)?

**Comment by dieterdreist** [comment]
2018-03-04 17:00 GMT+01:00 Christoph Hormann <notifications@github.com>:

> So @matthijsmelissen <https://github.com/matthijsmelissen> thinks this
> issue is solved, @kocio-pl <https://github.com/kocio-pl> does not
> understand it and @polarbearing <https://github.com/polarbearing> thinks
> it should not be solved.
>
> Just trying to understand the reasoning here.
>

I'm also for rendering landuse=military, i.e. agree the issue could be
closed

**Comment by imagico** [comment]
@dieterdreist - I was focusing on the more recent comments here - you were actually the only one who substantially argued the issue, with the idea that landuse=military implies access restrictions and rendering it in a way communicating access restrictions is therefore all right and that the breadth of meanings landuse=military is used for is no different from the scope of other tags like landuse=industrial or landuse=forest.
